### PR TITLE
Add ssh integration tests to ensure password is not in logs

### PR DESCRIPTION
**Pull request by Ch3LL**

### CODE REVIEW

**Comment by Ch3LL** [comment]
note: This will fail until this is merged https://github.com/saltstack/salt-ci-containers/pull/9 and container is re-generated

File: tests/pytests/integration/ssh/test_log.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            return True
+        except Exception as err:  # pylint: disable=broad-except
+            break
+        finally:
+            sock.close()
+        time.sleep(sleeptime)
+        sleeptime *= 2
+    else:
+        return False
+    return False
```
Comment (s0undt3ch) [inline tests/pytests/integration/ssh/test_log.py:42]:
```suggestion
def check_container_started(timeout_at, container):
    sleeptime = 2
    while time.time() <= timeout_at:
        time.sleep(sleeptime)
        try:
            if not container.is_running():
                return False
            sock = socket.socket()
            sock.connect(("localhost", container.get_check_ports()[22]))
            return True
        except Exception as err:  # pylint: disable=broad-except
            break
        finally:
            sock.close()
        time.sleep(sleeptime)
        sleeptime *= 2
    else:
        return False
    return False
```
Salt factories checks are not enough?

https://github.com/saltstack/pytest-salt-factories/blob/master/src/saltfactories/daemons/container.py#L656-L691

Comment (Ch3LL) [reply tests/pytests/integration/ssh/test_log.py:42]: I can probably remove this. Was a copy/paste. will remove

Comment (s0undt3ch) [reply tests/pytests/integration/ssh/test_log.py:42]: If we have this logic somewhere else, we should remove that too.

Comment (s0undt3ch) [reply tests/pytests/integration/ssh/test_log.py:42]: Not necessarily on this PR

Comment (Ch3LL) [reply tests/pytests/integration/ssh/test_log.py:42]: It honestly might actually just be solved by the extra time.sleep and is just good timing. Didn't realize that function was getting called. Let me dive in a bit more.

File: tests/pytests/integration/ssh/test_log.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+@pytest.fixture(scope="module")
+def ssh_docker_container(salt_factories, ssh_keys, ssh_auth):
+    """
+    Temporary docker container with python 3.6 and ssh enabled
+    """
+    ssh_pass, ssh_user = ssh_auth
+    container = salt_factories.get_container(
+        random_string("ssh-py_versions-"),
+        "ch3ll/salt-setup:ssh-password1",
```
Comment (waynew) [inline tests/pytests/integration/ssh/test_log.py:44]: **question** it's probably fine, at least for now, but is it possible to use the saltstack docker org or something?

**Review by waynew — DISMISSED** [review]
Just a comment, not a blocker. Looks good :+1:

File: tests/pytests/integration/ssh/test_log.py (hunk 2 above)
Comment (Ch3LL) [reply tests/pytests/integration/ssh/test_log.py:44]: Yeah, thanks for pointing that out. I meant to update it when it was available in our docker github. And now it is. Thanks!

File: tests/pytests/integration/ssh/test_log.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+def test_log_password(salt_ssh_cli, caplog, ssh_auth):
+    """
+    Test to ensure password is not logged when
+    using sudo and a password
+    """
+    ssh_pass, _ = ssh_auth
+    with caplog.at_level(logging.TRACE):
+        ret = salt_ssh_cli.run("--log-level=trace", "test.ping", minion_tgt="pyvertest")
+    if "kex_exchange_identification" in ret.stdout:
+        pytest.skip("Container closed ssh connection, skipping for now")
```
Comment (s0undt3ch) [inline tests/pytests/integration/ssh/test_log.py:119]: Why would we hit this? Do we run into the possibility of never running the test?

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**